deployment/common/changeset/set_config_mcms_test.go

```diff
@@ -0,0 +1,328 @@
+package changeset_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	commonchangeset "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+)
+
+// setupSetConfigTestEnv deploys all required contracts for the setConfig MCMS contract call.
+func setupSetConfigTestEnv(t *testing.T) deployment.Environment {
+
+	lggr := logger.TestLogger(t)
+	cfg := memory.MemoryEnvironmentConfig{
+		Nodes:  1,
+		Chains: 2,
+	}
+	env := memory.NewMemoryEnvironment(t, lggr, zapcore.DebugLevel, cfg)
+	chainSelector := env.AllChainSelectors()[0]
+	config := proposalutils.SingleGroupMCMS(t)
+
+	// Deploy MCMS and Timelock
+	env, err := commonchangeset.ApplyChangesets(t, env, nil, []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(commonchangeset.DeployLinkToken),
+			Config:    []uint64{chainSelector},
+		},
+		{
+			Changeset: commonchangeset.WrapChangeSet(commonchangeset.DeployMCMSWithTimelock),
+			Config: map[uint64]types.MCMSWithTimelockConfig{
+				chainSelector: {
+					Canceller:        config,
+					Bypasser:         config,
+					Proposer:         config,
+					TimelockMinDelay: big.NewInt(0),
+				},
+			},
```

Comment:
There is a [new util method](https://github.com/smartcontractkit/chainlink/blob/d0d9e8bbd8ee7c0c6cd054ca14ba1c3b77e8fcae/deployment/common/proposalutils/mcms_test_helpers.go#L85-L85) to simplify this

```suggestion
			Config: map[uint64]types.MCMSWithTimelockConfig{
				chainSelector: SingleGroupTimelockConfig(t),
			},
```

Reply:
nice thanks for sharing! added here https://github.com/smartcontractkit/chainlink/pull/15684/commits/d8cc495842e298e8a6b212fdb5c3bea4bd0929b5